mudlet-mapper.xml

```diff
@@ -1355,6 +1348,45 @@ mmp.locateAndEcho(room)</script>
                         <integer>1</integer>
                     </regexCodePropertyList>
                 </Trigger>
+                <Trigger isActive="yes" isFolder="no" isTempTrigger="no" isMultiline="no" isPerlSlashGOption="no" isColorizerTrigger="no" isFilterTrigger="no" isSoundTrigger="no" isColorTrigger="no" isColorTriggerFg="no" isColorTriggerBg="no">
+                    <name>Directory</name>
+                    <script>local dirnums = mmp.getnums(matches[4], true)
+local which
+
+if dirnums == nil then
+	echo(&quot; (?)&quot;)
+else
+	if #dirnums == 1 then
+		which = dirnums[1]
+		--echo(string.rep(&quot; &quot;, 30 - #matches[4]))
+		--cechoLink(&quot; - &lt;&quot;..mmp.echo_colour..&quot;&gt;&quot;..which, [[gotoRoom(]]..which..[[)]], &quot;Travel to &quot;..which, true)
+   	else
+		for index, val in ipairs(dirnums) do
+			if getRoomArea(val) == getRoomArea(mmp.currentroom) then which = val end
+		end
+		if which == nil then which = dirnums[1] end
```

Comment:
I'm not sure this should be the preferred solution. Your solution will
1. suggest going to the last room with that name in the area. Maybe you can try and filter for a room with the `shop` mark for the directory listing. But even then it's not necessarily unique and I'd inform the user about that.
2. If no room is found in the same area, then it suggests going to the first one in the list, no matter where that is. And no user can find out, where it'll take them if you give no indication. I'd also inform the user that no room in the current area is found.